src/form/lpac.jl

```diff
@@ -130,3 +130,254 @@ function constraint_ohms_yt_to(pm::AbstractLPACCModel, n::Int, f_bus, t_bus, f_i
+end
+
+""
+function variable_ne_branch_voltage_magnitude_to(pm::AbstractLPACModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    buses = ref(pm, nw, :bus)
+    branches = ref(pm, nw, :ne_branch)
+
+    phi_to_ne = var(pm, nw)[:phi_to_ne] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :ne_branch)], base_name="$(nw)_phi_to_ne",
+        start = comp_start_value(ref(pm, nw, :bus, branches[i]["t_bus"]), "phi_to_start")
+    )
+
+    if bounded
+        for (i, branch) in ref(pm, nw, :ne_branch)
+            JuMP.set_lower_bound(phi_to_ne[i], buses[branches[i]["t_bus"]]["vmin"] - 1.0)
+            JuMP.set_upper_bound(phi_to_ne[i], buses[branches[i]["t_bus"]]["vmax"] - 1.0)
+        end
+    end
+
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :phi_to, ids(pm, nw, :ne_branch), phi_to_ne)
+end
+
+""
+function variable_ne_branch_voltage_angle_fr(pm::AbstractLPACModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    buses = ref(pm, nw, :bus)
+    branches = ref(pm, nw, :ne_branch)
+
+    va_fr_ne = var(pm, nw)[:va_fr_ne] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :ne_branch)], base_name="$(nw)_va_fr_ne",
+        lower_bound = -2*pi,
+        upper_bound = 2*pi,
+        start = comp_start_value(ref(pm, nw, :bus, branches[i]["f_bus"]), "va_fr_start")
+    )
+
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :va_fr, ids(pm, nw, :ne_branch), va_fr_ne)
+end
+
+""
+function variable_ne_branch_voltage_angle_to(pm::AbstractLPACModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    buses = ref(pm, nw, :bus)
+    branches = ref(pm, nw, :ne_branch)
+
+    va_to_ne = var(pm, nw)[:va_to_ne] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :ne_branch)], base_name="$(nw)_va_to_ne",
+        lower_bound = -2*pi,
+        upper_bound = 2*pi,
+        start = comp_start_value(ref(pm, nw, :bus, branches[i]["t_bus"]), "va_to_start")
+    )
+
+    # may be bounds can be added in future
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :va_to, ids(pm, nw, :ne_branch), va_to_ne)
+end
+
+
+function variable_ne_branch_cosine(pm::AbstractLPACCModel; nw::Int=pm.cnw, report::Bool=true)
+
+    bi_bp = Dict((i, (b["f_bus"], b["t_bus"])) for (i,b) in ref(pm, nw, :ne_branch))
+    buspair = ref(pm, nw, :ne_buspairs)
+    cs_ne = var(pm, nw)[:cs_ne] = JuMP.@variable(pm.model,
+        [b in ids(pm, nw, :ne_branch)], base_name="$(nw)_cs_ne",
+        lower_bound = 0,
+        upper_bound = 1,
+        start = comp_start_value(ref(pm, nw, :ne_buspairs, bi_bp[b]), "cs_start", 1.0)
+    )
+
+    report && _IM.sol_component_value(pm, nw, :ne_branch, :cs_ne, ids(pm, nw, :ne_branch), cs_ne)
+end
+
+
+function constraint_ne_model_voltage(pm::AbstractLPACCModel, n::Int)
+    buses = ref(pm, n, :bus)
+    branches = ref(pm, n, :ne_branch)
+    bi_bp = Dict((i, (b["f_bus"], b["t_bus"])) for (i,b) in branches)
```

Comment:
Typically in NE models you cannot reason in terms of buspairs, becouse each line can be built individually you cannot reuse the variables that are shared between bus pairs.